src/pydvl/valuation/samplers/base.py

```diff
@@ -113,14 +112,33 @@ def from_data(self, data: Dataset) -> BatchGenerator:
 
     def from_indices(self, indices: IndexSetT) -> BatchGenerator:
         """Batches the samples and yields them."""
+
```

Comment:
If I see something like object.from_something(something), I would expect that another instance of the object's type is returned. But actually this generates batches, so my first impulse would be to call it generate_batches_from_indices(indices) and generate_batches(data). This is a discussion, we often have, but I am in the group "Clarity over brevity"

Reply:
As discussed, I removed `from_data` and renamed `from_indices` to `generate_batches`.